presto-parser/src/main/java/io/prestosql/sql/tree/Update.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class Update
+        extends Statement
+{
+    private final Table table;
+    private final List<Identifier> targets;
+    private final List<Expression> values;
+    private final Optional<Expression> where;
+
+    public Update(Table table, List<Identifier> targets, List<Expression> values, Optional<Expression> where)
+    {
+        this(Optional.empty(), table, targets, values, where);
+    }
+
+    public Update(NodeLocation location, Table table, List<Identifier> targets, List<Expression> values, Optional<Expression> where)
+    {
+        this(Optional.of(location), table, targets, values, where);
+    }
+
+    private Update(Optional<NodeLocation> location, Table table, List<Identifier> targets, List<Expression> values, Optional<Expression> where)
+    {
+        super(location);
+        this.table = requireNonNull(table, "table is null");
+        this.targets = requireNonNull(targets, "targets is null");
+        this.values = requireNonNull(values, "values is null");
+        checkArgument(targets.size() == values.size(), "targets (%s) and values (%s) have differents sizes", targets.size(), values.size());
```

Comment:
I would model them in a single list of "assignments". Then, you don't have to worry about whether the incoming lists have different numbers of elements.

Reply:
Indeed, this is much nicer!  Changed.